code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/TokenUtils.java

```diff
@@ -254,29 +257,42 @@ public String getAccessToken() {
 
     protected abstract String accessToken();
 
-    public Account getOrCreateAccount(Identity gotIdentity, Set<Identity> identities, Account account, boolean createAccount) {
-        boolean hasAccessToAProject = authDao.hasAccessToAnyProject(identities, false, null);
+    public Account getOrCreateAccount(Identity user, Set<Identity> identities, Account account) {
```

Comment:
The logic in this method was reworked so that the behavior across providers is the same. Meaning that account creation for a provider is specified with in the Implementation of token utils as dictated by createAccount(). As well as when the authentication is being turned on we record who turns it on, as well as instead of recycling the same admin account we create a new one for the user or use their existing one, unless the authentication mode they are using does not allow for creating an account in which case we throw an error. This case happens when local auth is being turned on and someone tries to create a token using a username and password that has not been associated with a pre existing account.